### DISCUSSION TITLE
ファームウェアの書き込みをしてもOLEDに「Keyball」というロゴが表示され続けキーボードとして認識されない。

### DISCUSSION THREAD

**Discussion by caGenki** [Yowkees/keyball#244] [Q&A] [answered]
# 困っていること

[Build Guide for Keyball61](https://github.com/Yowkees/keyball/blob/main/keyball61/doc/rev1/buildguide_jp.md)を参考に、先日遊舎工房さんが委託販売されていたKeyball61を組み立てています。
3-7にあるPro Microの書き込みと動作確認を行いましたが、キーボードとして認識されません。
似たようなdiscussionがないか検索しましたが、見当たらなかったため質問させていただきました。

# やったこと

1. MacBook ProでREMAPのサイトを開く。
2. Keyball61_testをFLASHをクリック。
3. USB Type-CケーブルでPro MicroとMacBook Proを接続。
4. connect選択ポップアップにて、RESETスイッチを2回押しして表示されたPro Microを選択。
5. FLASHをクリック。
6. `Writing the firmware finished successfully`と表示されたのを確認。
7. CLOSEをクリック。

このとき、OLEDに「Keyball」というロゴが表示され続けます。
テキストエディタなどを開いてキーの押下を試しましたが、文字が入力されることはありませんでした。
左右のキーボードそれぞれ同様にファームウェアの書き込みを行いましたが、どちらも同じ状態です。

![image](https://github.com/Yowkees/keyball/assets/103083534/da165517-abc5-41ad-8b3f-b89d6170fc85)
![image](https://github.com/Yowkees/keyball/assets/103083534/0484b86e-3ef2-4a74-bf86-a1f6385247b6)
![20230531_230913](https://github.com/Yowkees/keyball/assets/103083534/c6e491be-f8f6-421a-8275-6b49c1accf1f)

[REMAP](https://remap-keys.app/configure)で、キーマッピングができるか確認しましたが、キーボードとして認識されません。
RESETスイッチの2回押しを試しましたが、変化はありませんでした。

![image](https://github.com/Yowkees/keyball/assets/103083534/1ec69938-cbd4-478e-806d-75a8d4281961)

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。

testファームウェアではRemapが使用できませんのでviaファームウェアを書き込んでみてください。
また、USB-typeCコネクタは、すばやく奥まで差し込まないとUSBデバイスとして認識されないことがありますので、USBケーブルの抜き差しを試していただければ幸いです。

以上、お手数ですがご対応お願いいたします。

**Reply by caGenki** [reply]
お返事ありがとうございます。

解決しましたので報告いたします。

認識されなかった原因は、MacBook Proに直接接続していないことでした。
Thunderboltドックステーション経由ではなく、直接MacBook ProのThunderboltに接続したところ認識されるようになりました。

お騒がせして申し訳ありませんでした。

**Comment by Yowkees** [comment]
無事に解消して良かったです！安心しました。
ぜひ末永くご愛用いただければ幸いです。
